plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -3722,7 +3722,15 @@ protected GetVolumeStatsAnswer execute(GetVolumeStatsCommand cmd) {
                                 VirtualMachineMO vmMo = dcMo.findVm(file.getDir());
                                 Pair<VirtualDisk, String> vds = vmMo.getDiskDevice(file.getFileName(), true);
                                 long virtualsize = vds.first().getCapacityInKB() * 1024;
-                                long physicalsize = primaryStorageDatastoreMo.fileDiskSize(file.getPath());
+                                long physicalsize = 0L;
+                                VirtualMachineFileLayoutEx fileLayout = vmMo.getFileLayout();
+                                List<VirtualMachineFileLayoutExFileInfo> files = fileLayout.getFile();
+                                for(VirtualMachineFileLayoutExFileInfo fileInfo: files){
+                                    if (fileInfo.getName().contains("ROOT")){
```

Comment:
on second thought, there is no way here we know that we "only" want "ROOT" while iterating over a list of ```cmd.getVolumeUuids()```